### PR TITLE
test: additional tests for ssr

### PR DESCRIPTION
**Pull request by ravijayaramappa**
## Details

- [x]  if:true and if:false
- [ ] for:*
- [x] lwc:dom
- [x] lwc:dynamic

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

If yes, please describe the impact and migration path for existing applications.

## The PR fulfills these requirements:
* Have tests for the proposed changes been added? ✅ 
* Have you followed [these instructions](../CONTRIBUTING.md#-commit-message-conventions) to clearly describe the issue being fixed or feature enhanced? ✅ 

## GUS work item
W-7552627

### CODE REVIEW

File: packages/@lwc/engine-server/src/__tests__/fixtures/if-with-slots/modules/x/child/child.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<template></template>
```
Comment (pmdartus) [inline packages/@lwc/engine-server/src/__tests__/fixtures/if-with-slots/modules/x/child/child.html:1]: I think this template is missing a `<slot>`.

File: packages/@lwc/engine-server/src/__tests__/fixtures/if-conditional-slot-content/expected.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+              Should be slotted when false === false
+            </x-child>
+          </x-slotted>
+        </template>
+      </x-conditional-slot-content>
+    </div>
+    <div class="inactive-if-blocks">
+      <x-conditional-slot-content>
+        <template shadow-root>
+          <x-slotted>
```
Comment (ravijayaramappa) [inline packages/@lwc/engine-server/src/__tests__/fixtures/if-conditional-slot-content/expected.html:28]: notice that x-slotted has not light dom elements

File: packages/@lwc/engine-server/src/__tests__/fixtures/if-conditional-slot/expected.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            </x-child>
+          </x-slotted>
+        </template>
+      </x-parent>
+    </div>
+    <div class="inactive-if-blocks">
+      <x-parent>
+        <template shadow-root>
+          <x-slotted>
+            <template shadow-root></template>
```
Comment (ravijayaramappa) [inline packages/@lwc/engine-server/src/__tests__/fixtures/if-conditional-slot/expected.html:29]: `x-slotted`'s template has no `<slot>` elements

**Comment by pmdartus** [comment]
@ravijayaramappa Do you want to wait to add the `for:*` tests before merging, or do you want to proceed with what we have today?

**Comment by ravijayaramappa** [comment]
Will merge this and create a separate PR for for:* tests

**Pull request merged**